core/src/main/kotlin/org/evomaster/core/search/algorithms/AbstractGeneticAlgorithm.kt

```diff
@@ -24,6 +31,32 @@ abstract class AbstractGeneticAlgorithm<T> : SearchAlgorithm<T>() where T : Indi
     /** The current population of evaluated individuals (test suites). */
     protected val population: MutableList<WtsEvalIndividual<T>> = mutableListOf()
 
+    /**
+     * Frozen set of targets (coverage objectives) to score against during the current generation.
+     * Captured at the start of a generation and kept constant until the generation ends.
+     */
+    protected var frozenTargets: Set<Int> = emptySet()
+
+    @Inject
+    lateinit var selectionStrategy: SelectionStrategy
+
+    @Inject
+    lateinit var crossoverOperator: CrossoverOperator
+
+    @Inject
+    lateinit var mutationOperator: MutationOperator
```

Comment:
i don't think it is a good idea to use injection here. that is mainly done for singleton handling, which is not the case here.